src/org/opendatakit/briefcase/push/RemoteServer.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.opendatakit.briefcase.push;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import org.opendatakit.briefcase.reused.http.Credentials;
```

Comment:
What is the reason for the `reused` namespace?

Reply:
Yes, the nuance being that you don't start off by creating stuff on `reused`. You move stuff there from a business layer when you need to use it from another business layer, thus it becomes _reused_

Reply:
Is this part of some well-known software architecture scheme?